pkg/reconciler/awssnssource/subscription.go

```diff
@@ -82,11 +82,20 @@ func (r *Reconciler) ensureSubscribed(ctx context.Context) error {
 			return fmt.Errorf("%w", reconciler.NewEvent(corev1.EventTypeWarning, ReasonFailedSubscribe,
 				"Subscription is pending confirmation, will retry"))
 		case isAWSError(err):
-			// All documented API errors require some user intervention and
+			status.MarkNotSubscribed(v1alpha1.AWSSNSReasonRejected, "Subscription request rejected")
+			event := subscribeErrorEvent(url, topicARN, err)
+
+			// Most documented API errors require some user intervention and
 			// are not to be retried.
 			// https://docs.aws.amazon.com/sns/latest/api/API_Subscribe.html#API_Subscribe_Errors
-			status.MarkNotSubscribed(v1alpha1.AWSSNSReasonRejected, "Subscription request rejected")
-			return controller.NewPermanentError(subscribeErrorEvent(url, topicARN, err))
+			//
+			// However "InvalidParameter" is returned if the subscription URL is not ready, which
+			// might be the case if the endpoint for the source is not exposed yet.
+			if awsErr := awserr.Error(nil); errors.As(err, &awsErr) && awsErr.Code() != "InvalidParameter" {
```

Comment:
`sns.ErrCodeInvalidParameterException` can be used as the error code :slightly_smiling_face:

Reply:
thanks a lot!
fixing

Reply:
self review: also it should not be `!=` but `==` when the non permanent error is returned

Reply:
https://github.com/triggermesh/aws-event-sources/pull/290/commits/0d71f6a484f1cbea0e6f41f774b5edd95054fd5f